unipop-elastic/src/org/unipop/elastic/document/DocumentController.java

```diff
@@ -161,15 +164,34 @@ public Vertex addVertex(AddVertexQuery uniQuery) {
     //region Elastic Queries
 
     private <E extends Element, S extends DocumentSchema<E>> Iterator<E> search(SearchQuery<E> query, Map<S, Search> schemas) {
+        Optional<MutableMetrics> metrics = query.getStepDescriptor().getMetrics();
```

Comment:
please take all this new code out to some helper class. I can't see the search code from all the metrics code :)